### PR TITLE
✋ refactor: rpc service in actor

### PR DESCRIPTION
**Pull request by jjyr**
Use an event loop agent handle RPC calls,  to prevent holding lock too long.

### CODE REVIEW

File: rpc/src/agent.rs [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+            block_assembler,
+        }
+    }
+
+    #[allow(clippy::cyclomatic_complexity)]
+    pub fn start<S: ToString>(self, thread_name: Option<S>) -> RpcAgentController {
+        let (signal_sender, signal_receiver) =
+            crossbeam_channel::bounded::<()>(SIGNAL_CHANNEL_SIZE);
+        let (local_node_info_sender, local_node_info_receiver) =
+            crossbeam_channel::bounded(SIGNAL_CHANNEL_SIZE);
```
Comment (quake) [inline rpc/src/agent.rs]: should be `DEFAULT_CHANNEL_SIZE` ?

Comment (jjyr) [reply rpc/src/agent.rs]: Good catch

**Comment by keroro520** [comment]
Agent is a single point. It locks every time processing rpc-requests like before. So I think it has the same lock holding time with before.
I cannot figure out the point of why event loop agent prevent holding lock ?

And what about using RAII guard rule to optimize this problem ? @jjyr

**Comment by jjyr** [comment]
> Agent is a single point. It locks every time processing rpc-requests like before. So I think it has the same lock holding time with before.
> I cannot figure out the point of why event loop agent prevent holding lock ?
> 
> And what about using RAII guard rule to optimize this problem ? @jjyr

It can reduce the number of RPC lock waiters from n to 1 agent.

**Comment by doitian** [comment]
Need some benchmark to prove it worths to apply the changes.

**Comment by doitian** [comment]
Close for now.

**Pull request closed without merging**